### PR TITLE
Ajax-ified todo form: adding and deleting

### PR DESCRIPTION
**Pull request by jpalmieri**
Since I removed the todo#new route, all the feature specs that access this route are failing:

```
rspec ./spec/features/mark_todo_complete_spec.rb:5 # Delete todo item successfully
rspec ./spec/features/project_manager_creates_todo_spec.rb:10 # Project manager creates TODO Successfully
rspec ./spec/features/project_manager_creates_todo_spec.rb:19 # Project manager creates TODO Unsuccessfully: no description
rspec ./spec/features/todo_list_spec.rb:18 # View multiple todo items successfully
rspec ./spec/features/todo_list_spec.rb:26 # View multiple todo items without seeing '0' in 'days left' column
```

I'd like to get them passing with PhantomJS, but maybe we should merge this in first, and then start that on a new branch.

### CODE REVIEW

File: app/views/todos/index.html.erb [hunk 1]
```diff
@@ -15,12 +15,35 @@
...
+    </thead>
+  </table>
+<% end %>
+<%= form_for @todos.new, :html => {:class => "form-horizontal center"} do |form| %>
+  <div class="form-group">
+    <div class="col-md-10">
+      <%= form.text_field :description, autofocus: true, placeholder: 'Description', class: 'form-control' %>
+    </div>
+    <div class="col-md-2">
+      <%= form.submit 'Add Todo', class: 'btn btn-success', data: { :"todo-id" => @todos.new.id, :"create-todo-button" => true } %>
```
Comment (eliotsykes) [inline app/views/todos/index.html.erb]: Does the `data` options need `:"todo-id"` for this line?

File: app/views/todos/index.html.erb [hunk 2]
```diff
@@ -15,12 +15,35 @@
...
+        <td>#</td>
+        <td>Description</td>
+        <td>Days Left</td>
+        <td></td>
+      </tr>
+    </h4>
+    </thead>
+  </table>
+<% end %>
+<%= form_for @todos.new, :html => {:class => "form-horizontal center"} do |form| %>
```
Comment (eliotsykes) [inline app/views/todos/index.html.erb]: `:todo` could be used instead of `@todos.new` here for an admittedly very minor simplification

**Comment by eliotsykes** [comment]
I think it's OK to do the phantomjs work and editing the specs to get them passing with the moved "new form" as part of this branch. They are all part of the same ajaxification feature.

File: app/views/todos/index.html.erb (hunk 1 above)
Comment (jpalmieri) [inline app/views/todos/index.html.erb]: Ah, yeah. Thanks for catching that. Fixed in this commit: https://github.com/jpalmieri/blocitoff-api/commit/90e21fdd1fda62c83b16ca58b587b49a7b471f7f

File: app/views/todos/index.html.erb (hunk 2 above)
Comment (jpalmieri) [inline app/views/todos/index.html.erb]: Also fixed in this commit: https://github.com/jpalmieri/blocitoff-api/commit/90e21fdd1fda62c83b16ca58b587b49a7b471f7f

but I'm not clear on why using this symbol works.

**Comment by jpalmieri** [comment]
ok, cool. I may try to take a stab at it before then, but let's definitely work on this on Friday

File: app/views/todos/index.html.erb (hunk 2 above)
Comment (eliotsykes) [inline app/views/todos/index.html.erb]: Passing `form_for` a symbol is fine when we're not working with a persisted instance (as is the case when starting a new Todo) - Rails has enough information from the symbol name to figure out what model is needed to back this form (e.g. `:todo` -> `class Todo`).

The API docs go into a bit more detail on using symbols and using model instances: http://api.rubyonrails.org/classes/ActionView/Helpers/FormHelper.html#method-i-form_for

**Comment by eliotsykes** [comment]
> ok, cool. I may try to take a stab at it before then, but let's definitely work on this on Friday

Sounds good

**Comment by jpalmieri** [comment]
I just made a couple additional changes after the end of our meeting. If it looks good, I'll merge it and get it up on Heroku, then give you a demo

**Comment by jpalmieri** [comment]
btw, all specs are now passing

**Comment by jpalmieri** [comment]
Actually, I have a question before this is reviewed for a merge: Should I get rid of the `todos_controller.rb` methods and routes for `create` and `destroy`, and remove the corresponding tests, since they are no longer used? I'm not sure if it's conventional to leave them as a backup (I guess, in the case javascript fails). What are your thoughts on this?

**Comment by eliotsykes** [comment]
> Should I get rid of the todos_controller.rb methods and routes for create and destroy, and remove the corresponding tests, since they are no longer used? I'm not sure if it's conventional to leave them as a backup (I guess, in the case javascript fails). What are your thoughts on this?

Good thinking! I think leave them in. If JS fails, or is unavailable say due to a patchy mobile network, then yep it'd be good to have a backup, making your app more fault tolerant. 

Consider having scenarios in the feature specs to ensure the app has test coverage for the non-JS cases. In fact we had test coverage for some of these already before the `js: true` meta data was added in this branch.

By preserving the app's fundamental features to continue working without JS, you've followed a progressive enhancement approach to building your app. There's a good article on progressive enhancement [here](https://www.gov.uk/service-manual/making-software/progressive-enhancement.html).

**Comment by jpalmieri** [comment]
I didn't add extra specs for the js disabled experience, but I'm ok with that (if you're ok with that)

**Comment by eliotsykes** [comment]
Yep I'm happy with that.

**Pull request merged**